versioned-docs.md

```diff
@@ -0,0 +1,23 @@
+# Versioned Documentation
+
+We are starting to run into issues with people trying to use Fairlearn installed from PyPI with the documentation on [our website](https://fairlearn.github.io/).
+This is an issue because we have made and are making a number of breaking changes to our API.
+The solution to this is to provide versioned documentation, so that users can read the docs associated with the version they have installed.
+This is the pattern used by projects such as [SciKit-Learn](https://scikit-learn.org).
+
+Getting to something as sophisticated as the SciKit-Learn setup will take time.
+As a first step, we propose:
+
+1. Change the CirceCI integration to push to a `main` directory (in preparation for the expected rename of the `master` branch) within the [`fairlearn.github.io`](https://github.com/fairlearn/fairlearn.github.io) repository, rather than pushing to the root of the repository.
+
+1. Update links on the custom-designed landing page to go to this new `main` directory.
+
+1. Create a `v0.4.6` directory in `fairlearn.github.io`, and do a documentation build corresponding to the `v0.4.6` tag in this directory. This may well have to be done manually.
```

Comment:
Everything is built from the fairlearn repo, so we wouldn't manually create these directories, but rather instruct our pipeline to put the results of the builds there. We should keep treating the website repo as something we can delete at any point and it gets fully recreated with the next push.

Reply:
Problem then is getting the v0.4.6 documentation (although your point about automation is well taken). Right now, a huge pain point is that people install from PyPI, and then try reading the docs here.

If we're willing to leave that problem to one side (in the hopes of v0.4.7 coming out soon after whatever work this requires), then full automation is certainly an option.

Reply:
I'm not sure I understand. This problem is unrelated to any particular version since we just need to start at some point, right?

Reply:
It's just that v0.4.6 is a particular problem right now. Having said that, the packages you listed below do appear to support pattern matching on tags and the like, so that might be able to do the necessary.